hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -262,6 +268,24 @@ public void start(OzoneConfiguration configuration) {
         snapshotSstFilteringService.start();
       }
     }
+
+    if (snapshotDeletingService == null) {
+      long snapshotServiceInterval = configuration.getTimeDuration(
+          OZONE_SNAPSHOT_DELETION_SERVICE_INTERVAL,
+          OZONE_SNAPSHOT_DELETION_SERVICE_INTERVAL_DEFAULT,
+          TimeUnit.MILLISECONDS);
+      long snapshotServiceTimeout = configuration.getTimeDuration(
+          OZONE_SNAPSHOT_DELETION_SERVICE_TIMEOUT,
+          OZONE_SNAPSHOT_DELETION_SERVICE_TIMEOUT_DEFAULT,
+          TimeUnit.MILLISECONDS);
+      try {
+        snapshotDeletingService = new SnapshotDeletingService(
+            snapshotServiceInterval, snapshotServiceTimeout, ozoneManager);
+        snapshotDeletingService.start();
```

Comment:
So this should only run on the leader OM right?

Reply:
Yes it only runs on leader OM, the check is done here `shouldRun()`